From 254b742c6431dd80d146a91eab29677af4998144 Mon Sep 17 00:00:00 2001
From: Yann Dirson <ydirson@free.fr>
Date: Sat, 1 Mar 2014 00:38:41 +0100
Subject: [PATCH] Fix install rule for packaging.

* $(DESTDIR) must be used everywhere
* /usr/games/ may not exist
* DATADIR ought to be easy to override when autodetection is not needed,
  to avoid build-dependency on xboard
---
 Makefile | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index b76476d..eafa108 100644
--- a/Makefile
+++ b/Makefile
@@ -3,6 +3,7 @@ srcdir = .
 CC?=gcc
 CFLAGS?= -O2 -s
 
+DATADIR=`xboard --show-config Datadir`
 
 ALL= hachu hachu.6.gz
 
@@ -12,13 +13,14 @@ hachu: hachu.c
 	$(CC) $(CFLAGS) hachu.c -o hachu
 
 install: ${ALL} ${srcdir}/svg/*
+	install -d -m0755 $(DESTDIR)/usr/games
 	cp -u ${srcdir}/hachu $(DESTDIR)/usr/games
 	install -d -m0755 $(DESTDIR)/usr/share/man/man6
 	cp -u ${srcdir}/hachu.6.gz $(DESTDIR)/usr/share/man/man6
-	install -d -m0755 `xboard --show-config Datadir`/themes/chu
-	cp -u ${srcdir}/svg/*.svg `xboard --show-config Datadir`/themes/chu
-	install -d -m0755 `xboard --show-config Datadir`/themes/conf
-	cp -u ${srcdir}/svg/sho ${srcdir}/svg/chu `xboard --show-config Datadir`/themes/conf
+	install -d -m0755 $(DESTDIR)$(DATADIR)/themes/chu
+	cp -u ${srcdir}/svg/*.svg $(DESTDIR)$(DATADIR)/themes/chu
+	install -d -m0755 $(DESTDIR)$(DATADIR)/themes/conf
+	cp -u ${srcdir}/svg/sho ${srcdir}/svg/chu $(DESTDIR)$(DATADIR)/themes/conf
 
 hachu.6.gz: hachu.pod
 	pod2man -s 6 hachu.pod > hachu.man
-- 
1.8.5.3

